java/code/src/com/suse/manager/webui/services/SaltServerActionService.java

```diff
@@ -2500,11 +2500,14 @@ public void executeSSHAction(Action action, MinionServer minion) {
                 return;
             }
 
-            sa.setRemainingTries(sa.getRemainingTries() - 1);
-
             Map<LocalCall<?>, List<MinionSummary>> calls = callsForAction(action,
                     Arrays.asList(new MinionSummary(minion)));
 
+            sa.setRemainingTries(sa.getRemainingTries() - 1);
+            sa.setStatus(ActionFactory.STATUS_PICKED_UP);
+            sa.setPickupTime(new Date());
+            HibernateFactory.commitTransaction();
```

Comment:
`commit` is not so good, but I think in this case needed. I think SSH keeps the connection open until the result is there. To switch the state 2 times and make it visible to others, we have to commit.
I am right?

Reply:
Exactly, it waits until the whole thing is finished otherwise. I tested it manually with this, and everything works, but @rjmateus could maybe say more.

Reply:
And tracing back the code, there's no other write operation happening before, so in this specific case, we should be safe.

Reply:
I get your point, and we don't have many options here. For the state to be visible to others we need to commit to the database.
One option could be open a new DB transaction to commit just this change in the action state, but I don't think that effort is needed here.

I have just one question, why are the state change and commit after the `callsForAction` method call?
In my opinion we should do this just after making the decision we will be processing this action, to avoid more code that could be interacting with the database and be affected by this commit. Could these changes be moved to line 2502, just after the last check?

Reply:
> In my opinion we should do this just after making the decision we will be processing this action, to avoid more code that could be interacting with the database and be affected by this commit. Could these changes be moved to line 2502, just after the last check?

Considering this, and @mcalmer's question about the unit test safety, I'll move this to an outer class, to `SSHMinionActionExecutor#execute`. This part is never called from unit tests directly, so it would be safer for the future as well.